model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -43,97 +43,101 @@ async def generate(request: Request) -> Response:
     # check health before accepting request and fail fast if engine isn't healthy
     try:
         await engine.check_health()
-    except Exception as e:
-        print(f"The vllm engine is dead, exiting the pod: {e}")
-        os.kill(os.getpid(), signal.SIGINT)
 
-    request_dict = await request.json()
-    prompt = request_dict.pop("prompt")
-    stream = request_dict.pop("stream", False)
-    guided_json = request_dict.pop("guided_json", None)
-    guided_regex = request_dict.pop("guided_regex", None)
-    guided_choice = request_dict.pop("guided_choice", None)
-    guided_grammar = request_dict.pop("guided_grammar", None)
-    sampling_params = SamplingParams(**request_dict)
+        request_dict = await request.json()
```

Comment:
what are these changes?

Reply:
i thought your exception handling change is already merged?

Reply:
I wrapped the entire function in the try: catch. My previous fix would kill the process on a new request, but his actually kills the process on the request that crashed the vllm engine